src/main/java/org/broadinstitute/hellbender/engine/filters/CountedReadFilter.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+
+/**
+ * Wrapper/adapter for ReadFilter that counts the number of reads filtered, and provides a filter count summary.
+ *
+ * For filters with complex predicates that are composed of compound and/or operators, provides counts
+ * by level for each component predicate.
+ *
+ */
+public class CountedReadFilter implements ReadFilter {
+
+    static final long serialVersionUID = 1L;
+
+    // Underlying ReadFilter we delegate to if we're wrapping a simple ReadFilter
+    protected final ReadFilter delegateFilter;
+
+    // Name for this filter, used to report filter count summaries
+    protected final String name;
+
+    // Number of reads filtered by this filter
+    protected long filteredCount = 0;
+
+    public CountedReadFilter(final String filterName, final ReadFilter readFilter) {
+        name = filterName;
+        delegateFilter = readFilter;
+    }
+
+    private CountedReadFilter(final String filterName) {
+        name = filterName;
+        delegateFilter = null;
+    }
+
+    // Return the number of reads filtered by this filter
+    public long getFilteredCount() {
+        return filteredCount;
+    }
+
+    public String getName() {return name;}
+
+    // Returns a summary line with filter counts organized by level
+    public String getSummaryLine() {return getSummaryLineForLevel(0);}
+
+    protected String getSummaryLineForLevel(final int indentLevel) {
+        return filteredCount == 0 ?
+                "" :
+                getIndentString(indentLevel) + Long.toString(filteredCount) + " read(s) filtered by: " + getName() + " \n";
+    }
+
+    protected String getIndentString(final int indentLevel) {
+        final StringBuilder bldr = new StringBuilder();
+        IntStream.range(0, indentLevel).forEach(i -> bldr.append("  "));
+        return bldr.toString();
+    }
+
+    /**
+     * Specialization of {@link #and(Predicate)} so that CountedReadFilter and'ed with other CountedReadFilter produce a CountedReadFilter
+     */
+    //@Override
+    public CountedReadFilter and(final CountedReadFilter other) {
+        Objects.requireNonNull(other);
```

Comment:
Use `Utils.nonNull()` here and below, which throws the less-alarming `IllegalArgumentException` rather than `NullPointerException`